timemachine/fe/rbfe.py

```diff
@@ -1,101 +1,109 @@
 from timemachine.md import builders, minimizer
 from timemachine.md.barostat.utils import get_bond_list, get_group_indices
 from timemachine.parallel.client import AbstractClient, AbstractFileClient, CUDAPoolClient, FileClient
 from timemachine.potentials import jax_utils
 
 
-def get_batch_U_fns(bps):
-    # return a function that takes in coords, boxes
-    all_U_fns = []
-    for bp in bps:
+def setup_host(st: SingleTopology, host_config: HostConfig):
 
-        def batch_U_fn(xs, boxes, bp_impl):
-            Us = []
-            for x, box in zip(xs, boxes):
-                # tbd optimize to "selective" later
-                _, U = bp_impl.execute(x, box)
-                Us.append(U)
-            return np.array(Us)
+    host_system, host_masses = convert_omm_system(host_config.omm_system)
+    host_conf = minimizer.minimize_host_4d(
+        [st.mol_a, st.mol_b],
+        host_config.omm_system,
+        host_config.conf,
+        st.ff,
+        host_config.box,
+    )
+    host = (host_system, host_masses, host_conf)
+    return host
 
-        # extra functools.partial is needed to deal with closure jank
-        all_U_fns.append(functools.partial(batch_U_fn, bp_impl=bp))
 
-    return all_U_fns
+def setup_in_vacuum(st, ligand_conf, lamb):
 
+    system = st.setup_intermediate_state(lamb)
+    combined_masses = np.array(st.combine_masses())
 
-def sample(initial_state, protocol):
-    """
-    Generate a trajectory given an initial state and a simulation protocol
-    """
+    potentials = system.get_U_fns()
+    hmr_masses = model_utils.apply_hmr(combined_masses, system.bond.get_idxs())
+    baro = None
 
-    bound_impls = [p.bound_impl(np.float32) for p in initial_state.potentials]
-    intg_impl = initial_state.integrator.impl()
-    if initial_state.barostat:
-        baro_impl = initial_state.barostat.impl(bound_impls)
-    else:
-        baro_impl = None
-
-    ctxt = custom_ops.Context(
-        initial_state.x0,
-        initial_state.v0,
-        initial_state.box0,
-        intg_impl,
-        bound_impls,
-        baro_impl,
-    )
+    x0 = ligand_conf
+    box0 = np.eye(3, dtype=np.float64) * 10  # make a large 10x10x10nm box
 
-    # burn-in
-    ctxt.multiple_steps_U(
-        n_steps=protocol.n_eq_steps,
-        store_u_interval=0,
-        store_x_interval=0,
-    )
+    return x0, box0, hmr_masses, potentials, baro
 
-    assert np.all(np.isfinite(ctxt.get_x_t())), "Equilibration resulted in a nan"
 
-    # a crude, and probably not great, guess on the decorrelation time
-    n_steps = protocol.n_frames * protocol.steps_per_frame
-    _, all_coords, all_boxes = ctxt.multiple_steps_U(
-        n_steps=n_steps,
-        store_u_interval=0,
-        store_x_interval=protocol.steps_per_frame,
-    )
+def setup_in_env(st, host, host_config, ligand_conf, lamb, temperature, run_seed):
+
+    host_system, host_masses, host_conf = host
+
+    # minimize water box around the ligand by 4D-decoupling
```

Comment:
Is this comment correct? Doesn't look like `combine_with_host` does any minimization and also this may not just be a water box.

Reply:
This comment should probably be moved closer to `optimize_coordinates` -- previously was here: https://github.com/proteneer/timemachine/blob/6aa9edc08dbe000ab315aa85f03a8de42f06faec/timemachine/fe/rbfe.py#L180

Reply:
Updated in https://github.com/proteneer/timemachine/pull/936/commits/5b65ddf29bcefbdc83ac6dc9279ef769a05e3f12